include/dca/phys/dca_step/cluster_solver/ctint/walker/ctint_walker_base.hpp

```diff
@@ -279,6 +298,9 @@ void CtintWalkerBase<Parameters, Real>::markThermalized() {
   sign_avg_.reset();
   n_accepted_ = 0;
   n_steps_ = 0;
+
+  // Recompute the Monte Carlo weight.
+  setMFromConfig();
```

Comment:
I am recomputing the weight from scratches here as we started from a random config, to avoid numerical issues with an initial weight that is likely to be very close to zero.